socorro/lib/task_manager.py

```diff
@@ -102,13 +98,10 @@ def _get_iterator(self):
         This sorts that out and returns an iterator.
 
         """
-        try:
-            return self.job_param_source_iter(self.config)
-        except TypeError:
-            try:
-                return self.job_param_source_iter()
-            except TypeError:
-                return self.job_param_source_iter
+        # FIXME(willkg): test this out; we dropped the class instance one
+        if callable(self.job_source_iterator):
+            return iter(self.job_source_iterator())
+        return iter(self.job_source_iterator)
```

Comment:
I rewrote this, but I'm not sure it does the same thing it used to do because the old code is bereft of "this is what this is intended to do" comments. I redid the related tests as well and it works with the code we have in Socorro, so it's probably fine.